### PR TITLE
Have ntp sync up waiting for 1 minute in case sync up failed

### PR DESCRIPTION
**Pull request by zet809**
Fix issue #2327 
To have ntp wait for 1 minutes if sync up failed.

### CODE REVIEW

**Comment by whowutwut** [comment]
Have we run this code on a cluster and do we have the output that shows the behavior change before and after?  I didn't capture the time stamp in the issue before. Will check the console log when I get to the office if there still is the history

File: xCAT-genesis-scripts/bin/doxcat [hunk 1]
```diff
@@ -267,7 +267,8 @@ logger -s -t $log_label -p local4.info "Starting ntpd..."
 ntpd -g -x
 
 logger -s -t $log_label -p local4.info "Waiting for ntpd to synchronize..."
-ntp-wait -n 1999 -s 1 -v
+# Wait 1 minute in case ntp sync up failed
+ntp-wait -n 10 -v
```
Comment (whowutwut) [inline xCAT-genesis-scripts/bin/doxcat]: Can we make this a variable and print out the time in the logger msg at line 269?

**Comment by whowutwut** [comment]
I tried it out and it time'ed out after 1 minute... 
```
<166>Jan  3 21:20:18 xcat.genesis.doxcat: the nic enP3p3s0f0 can ping 50.3.27.2
<166>Jan  3 21:20:21 xcat.genesis.doxcat: Acquired IPv4 address on enP3p3s0f0
50.3.27.102/8
<166>Jan  3 21:20:21 xcat.genesis.doxcat: Starting ntpd...
<166>Jan  3 21:20:21 xcat.genesis.doxcat: Waiting for ntpd to synchronize...
Waiting for ntpd to synchronize... No!
ntpd did not synchronize.
<166>Jan  3 21:21:21 xcat.genesis.doxcat: ... ntpd did not synchronize.
<166>Jan  3 21:21:21 xcat.genesis.doxcat: Attempting to sync hardware clock...
<166>Jan  3 21:21:21 xcat.genesis.doxcat: Restarting syslog...
```

Can we

File: xCAT-genesis-scripts/bin/doxcat (hunk 1 above)
Comment (whowutwut) [inline xCAT-genesis-scripts/bin/doxcat]: We also noticed that the spinning icon is slower,  I think this is caused by changing the `-s 1` to default to 6 seconds..  Did you intentionally want to make this change?

**Comment by whowutwut** [comment]
Hi @zet809  If you can answer the questions in the review, we can get this resolved and merged.  Thanks!

File: xCAT-genesis-scripts/bin/doxcat (hunk 1 above)
Comment (zet809) [inline xCAT-genesis-scripts/bin/doxcat]: Hi, @whowutwut , I think it is ok to me, I changed the interval to be 1 second.

**Comment by zet809** [comment]
hi, @whowutwut , I had updated the modificaiton based on your comment, pls help to review. Thx a lot!

**Comment by whowutwut** [comment]
@zet809  I think once a review started, we should avoid the `--amend` option for the commit.  It's hard for me to see the past changes.

File: xCAT-genesis-scripts/bin/doxcat [hunk 2]
```diff
@@ -266,8 +266,9 @@ ip addr show dev $bootnic|grep -v 'scope link'|grep -v 'dynamic'|grep -v  inet6|
 logger -s -t $log_label -p local4.info "Starting ntpd..." 
 ntpd -g -x
 
-logger -s -t $log_label -p local4.info "Waiting for ntpd to synchronize..."
-ntp-wait -n 1999 -s 1 -v
+NTP_WAIT_SECONDS=60
+logger -s -t $log_label -p local4.info "Waiting $NTP_WAIT_SECONDS seconds for ntpd to synchronize..."
+ntp-wait -n $NTP_WAIT_SECONDS -s 1 -v
```
Comment (whowutwut) [inline xCAT-genesis-scripts/bin/doxcat]: Sorry for this back and forth...  @zet809  I read the man page more carefully now and seems like ntp-wait defaults to 6 seconds.   @neo954 had changed it to -s 1 in the initial code so I was just questioning whether you intentionally made that change...  I didn't read the man page.  

But now, wondering if we should change to cause more attempts to ping the ntp server or just leave it as the default to spread out the impact if we have hundreds of nodes coming up at once... 

Suggest:
```
# ntp-wait defaults to 6 seconds between retries, wait for 1 minute
NTP_TRIES=10
NTP_SLEEP=6
logger -s -t $log_label -p local4.info "Waiting for $NTP_TRIES x $NTP_SLEEP seconds for ntpd to synchronize..."
ntp-wait -n $NTP_TRIES -s $NTP_SLEEP -v
```

The only reason is that this makes it clear in the event that users want to adjust the wait time and rerun `mknb`...

Comment (zet809) [inline xCAT-genesis-scripts/bin/doxcat]: Thx, @whowutwut , I had modified the code based on your comments, pls review.

**Comment by whowutwut** [comment]
Let's merge this, will test this out in discovery this week.

**Comment by whowutwut** [comment]
Verified this today
```
[Tue Mar 21 22:57:54 2017]<166>Jan 12 04:26:15 xcat.genesis.doxcat: Starting ntpd...
[Tue Mar 21 22:57:54 2017]<166>Jan 12 04:26:15 xcat.genesis.doxcat: Waiting for 10 x 6 seconds for ntpd to synchronize...
[Tue Mar 21 22:57:54 2017]Waiting for ntpd to synchronize... No!
[Tue Mar 21 22:58:54 2017]ntpd did not synchronize.
[Tue Mar 21 22:58:54 2017]<166>Jan 12 04:27:15 xcat.genesis.doxcat: ... ntpd did not synchronize.
[Tue Mar 21 22:58:54 2017]<166>Jan 12 04:27:15 xcat.genesis.doxcat: Attempting to sync hardware clock...
[Tue Mar 21 22:58:54 2017]<166>Jan 12 04:27:15 xcat.genesis.doxcat: Restarting syslog...
```

**Comment by daniceexi** [comment]
👍

**Pull request merged**